vmm/src/config.rs

```diff
@@ -2193,6 +2198,12 @@ impl VmConfig {
             }
         }
 
+        if let Some(vsock) = &self.vsock {
+            if [u32::MAX as u64, 0, 1, 2].contains(&vsock.cid) {
```

Comment:
The tests rely on being able to use `VsockConfig::default`.  I could modify the default value to use CID 3 or something, but I think it really doesn't make sense for structs like VsockConfig to implement default.  There's no sensible default value for a CID, or especially for a socket path.  I've tried to make some progress towards removing the need for every config struct to implement a default value, even where it doesn't make sense, in #6081.

Reply:
Can you not modify the tests not to use `VsockConfig::default()` - but otherwise it seems reasonable to set an appropriate default - we don't want perfection to be the enemy of better - and it's definitely better to reject invalid CIDs.

Reply:
Done.